sklearn/experimental/_column_transformer.py

```diff
@@ -0,0 +1,446 @@
+        """Fit all transformers, transform the data and concatenate results.
+
+        Parameters
+        ----------
+        X : array-like or DataFrame of shape [n_samples, n_features]
+            Input data, of which specified subsets are used to fit the
+            transformers.
+
+        y : array-like, shape (n_samples, ...), optional
+            Targets for supervised learning.
+
+        Returns
+        -------
+        X_t : array-like or sparse matrix, shape (n_samples, sum_n_components)
+            hstack of results of transformers. sum_n_components is the
+            sum of n_components (output dimension) over transformers. If
+            one result is a sparse matrix, everything will be converted to
+            sparse matrices.
+
+        """
+        self._validate_transformers()
+        try:
+            result = Parallel(n_jobs=self.n_jobs)(
+                delayed(_fit_transform_one)(clone(trans), weight, X_sel, y,
+                                            **fit_params)
+                for name, trans, X_sel, weight in self._iter(X=X))
+        except ValueError as e:
+            if "Expected 2D array, got 1D array instead" in str(e):
+                raise ValueError(_ERR_MSG_1DCOLUMN)
+            else:
+                raise
+
+        if not result:
+            # All transformers are None
+            return np.zeros((X.shape[0], 0))
+        Xs, transformers = zip(*result)
+        self._update_fitted_transformers(transformers)
+        if any(sparse.issparse(f) for f in Xs):
+            Xs = sparse.hstack(Xs).tocsr()
+        else:
+            Xs = np.hstack(Xs)
+        return Xs
+
+    def transform(self, X):
+        """Transform X separately by each transformer, concatenate results.
+
+        Parameters
+        ----------
+        X : array-like or DataFrame of shape [n_samples, n_features]
+            Input data, of which specified subsets are used to fit the
+            transformers.
+
+        Returns
+        -------
+        X_t : array-like or sparse matrix, shape (n_samples, sum_n_components)
+            hstack of results of transformers. sum_n_components is the
+            sum of n_components (output dimension) over transformers. If
+            one result is a sparse matrix, everything will be converted to
+            sparse matrices.
+
+        """
+        check_is_fitted(self, 'transformers_')
+        try:
+            Xs = Parallel(n_jobs=self.n_jobs)(
+                delayed(_transform_one)(trans, weight, X_sel)
+                for name, trans, X_sel, weight in self._iter(X=X, fitted=True))
+        except ValueError as e:
+            if "Expected 2D array, got 1D array instead" in str(e):
+                raise ValueError(_ERR_MSG_1DCOLUMN)
+            else:
+                raise
+
+        if not Xs:
+            # All transformers are None
+            return np.zeros((X.shape[0], 0))
+        if any(sparse.issparse(f) for f in Xs):
```

Comment:
There is the same code in `transform` and `fit_transform`. You might want to create a function.

```python
     if any(sparse.issparse(f) for f in Xs):
           Xs = sparse.hstack(Xs).tocsr()
      else:
           Xs = np.hstack(Xs)
      return Xs
```